### ISSUE TITLE
No attribute 'load_state_dict' when running inference

### ISSUE THREAD

**Issue by andrewsimonds14** [MIC-DKFZ/nnUNet#563]
Hi,
I'm currently testing out a model I trained to run inference on another .nii file.

When I run this command: nnUNet_predict -i ./inputTest -o ./outputTest -t 502 -m 3d_fullres --save_npz

The preprocessing steps seem to complete fine, but it runs into an issue with the actual prediction.
The contents of ./inputTest is the one .nii.gz folder and ./outputTest is just empty for now, but will be where the mask is saved.

Here is the terminal output after running the command:
nnUNet_predict -i ./inputTest -o ./outputTest -t 502 -m 3d_fullres --save_npz

Please cite the following paper when using nnUNet:

Isensee, F., Jaeger, P.F., Kohl, S.A.A. et al. "nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation." Nat Methods (2020). https://doi.org/10.1038/s41592-020-01008-z

If you have questions or suggestions, feel free to open an issue at https://github.com/MIC-DKFZ/nnUNet

using model stored in  /home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1
This model expects 1 input modalities for each image
Found 1 unique case ids, here are some examples: ['FSRTCASE_001']
If they don't look right, make sure to double check your filenames. They must end with _0000.nii.gz etc
number of cases: 1
number of cases that still need to be predicted: 1
emptying cuda cache
loading parameters for folds, None
folds is None so we will automatically look for output folders (not using 'all'!)
found the following folds:  ['/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_0', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_1', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_3', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_4']
using the following model files:  ['/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_0/model_final_checkpoint.model', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_1/model_final_checkpoint.model', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_3/model_final_checkpoint.model', '/home/lab/nnUNet_data/RESULTS_FOLDER/nnUNet/3d_fullres/Task502_BrainMetastases/nnUNetTrainerV2__nnUNetPlansv2.1/fold_4/model_final_checkpoint.model']
starting preprocessing generator
starting prediction...
preprocessing ./outputTest/FSRTCASE_001.nii.gz
using preprocessor GenericPreprocessor
before crop: (1, 208, 256, 200) after crop: (1, 208, 255, 199) spacing: [1. 1. 1.] 

no resampling necessary
no resampling necessary
before: {'spacing': array([1., 1., 1.]), 'spacing_transposed': array([1., 1., 1.]), 'data.shape (data is transposed)': (1, 208, 255, 199)} 
after:  {'spacing': array([1., 1., 1.]), 'data.shape (data is resampled)': (1, 208, 255, 199)} 

(1, 208, 255, 199)
This worker has ended successfully, no errors to report
predicting ./outputTest/FSRTCASE_001.nii.gz
Traceback (most recent call last):
  File "/home/lab/anaconda3/bin/nnUNet_predict", line 33, in <module>
    sys.exit(load_entry_point('nnunet', 'console_scripts', 'nnUNet_predict')())
  File "/home/lab/nnUNet/nnunet/inference/predict_simple.py", line 217, in main
    predict_from_folder(model_folder_name, input_folder, output_folder, folds, save_npz, num_threads_preprocessing,
  File "/home/lab/nnUNet/nnunet/inference/predict.py", line 630, in predict_from_folder
    return predict_cases(model, list_of_lists[part_id::num_parts], output_files[part_id::num_parts], folds,
  File "/home/lab/nnUNet/nnunet/inference/predict.py", line 214, in predict_cases
    trainer.load_checkpoint_ram(p, False)
  File "/home/lab/nnUNet/nnunet/training/network_training/network_trainer.py", line 362, in load_checkpoint_ram
    self.amp_grad_scaler.load_state_dict(checkpoint['amp_grad_scaler'])
AttributeError: 'NoneType' object has no attribute 'load_state_dict'

Let me know if you have any suggestions or need more info to help!
Thanks

**Comment by FabianIsensee** [comment]
Hi,
could it be that you are running prediction in FP32 while training was in FP16? I believe this issue is fixed in the current master. Could you try that and report whether that solved it for you?
Best,
Fabain

**Comment by andrewsimonds14** [comment]
Hey,
I think that solved the issue!  Now I have some generic CUDA issues but should be good to go, thanks for your help!
Andrew

**Comment by FabianIsensee** [comment]
glad to hear it works now. Upgrading to the most recent master could also solve this issue.
Best
Fabian

**Comment by karthikk2085** [comment]
Hi Fabian,

I am currently testing a model from the best model version by copying the same file in the same folder and renaming the copied file version to "model_final_checkpoint.model" to see if I can any inference using these files. When I tried to run on a fold, I'm facing the similar error("AttributeError: 'NoneType' object has no attribute 'load_state_dict'") even though I have updated the repo with the latest master. Am I missing anything ? Should I wait for the whole training to get completed ?

Would really appreciate your help. 

Thanks
Karthik
